src/System.Runtime.Extensions/src/System/IO/StreamReader.cs

```diff
@@ -39,7 +32,6 @@ public class StreamReader : TextReader
         private Decoder _decoder;
         private byte[] _byteBuffer;
         private char[] _charBuffer;
-        private byte[] _preamble;   // Encoding's preamble, which identifies this encoding.
```

Comment:
I think you're right that the 99.999% case will be the built-in encodings, and for those this change makes sense.  Let's do it.